File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int cnt = 1;
        while(cnt <= n){ 
            int sum = 0;
            for(int i = cnt; i <= n; i++) {
               sum += i;
                if(sum == n){
                    answer++;
                    cnt++;
                    break;
                } else if(sum > n){
                    cnt++;
                    break;
                }
            
            } 
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {    
    const answer = [];
    let numbers = [];
    let i = 1,
        j = 1;

    while(i <= n){
        if(i > n) break;        
        numbers.push(i);
        let total = 0;
        for(const num of numbers) 
            total += num;
        if(total === n){ 
            answer.push(numbers.shift());                 
            i = j++;             
        }else if(total > n){
            numbers = [];
            i = j++;
        }        
        i++;
    }

    return answer.length;
}